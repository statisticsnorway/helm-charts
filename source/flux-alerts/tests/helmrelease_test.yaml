suite: test helmrelease for backend with minimal config

tests:
  - it: should render Alert
    templates:
      - alert.yaml
    values:
      - ./values/helmrelease.yaml
    release:
      name: flux-alert
      namespace: unittest
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Alert
      - matchSnapshot: {}

  - it: should render Provider
    templates:
      - provider.yaml
    values:
      - ./values/helmrelease.yaml
    release:
      name: flux-alert
      namespace: unittest
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Provider
      - matchSnapshot: {}

  - it: should render SealedSecret
    templates:
      - sealedsecret.yaml
    values:
      - ./values/helmrelease.yaml
    release:
      name: flux-alert
      namespace: unittest
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: SealedSecret
      - matchSnapshot: {}
