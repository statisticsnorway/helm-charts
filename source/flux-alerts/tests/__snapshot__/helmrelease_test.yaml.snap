should render Alert:
  1: |
    apiVersion: notification.toolkit.fluxcd.io/v1beta2
    kind: Alert
    metadata:
      labels:
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/version: "1.0"
        helm.sh/chart: flux-alerts-1.0.2
      name: flux-alert-flux-alerts
    spec:
      eventSeverity: warning
      eventSources:
      - kind: GitRepository
        name: '*'
      exclusionList:
      - Exclude this message
      providerRef:
        name: flux-alert-flux-alerts
      summary: Alert from test-cluster
should render Provider:
  1: |
    apiVersion: notification.toolkit.fluxcd.io/v1beta2
    kind: Provider
    metadata:
      labels:
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/version: "1.0"
        helm.sh/chart: flux-alerts-1.0.2
      name: flux-alert-flux-alerts
    spec:
      address: https://slack.com/api/chat.postMessage
      channel: test-channel
      secretRef:
        name: flux-alert-flux-alerts
      type: slack
should render SealedSecret:
  1: |
    apiVersion: bitnami.com/v1alpha1
    kind: SealedSecret
    metadata:
      annotations:
        sealedsecrets.bitnami.com/cluster-wide: "true"
      labels:
        app.kubernetes.io/managed-by: Tiller
        app.kubernetes.io/version: "1.0"
        helm.sh/chart: flux-alerts-1.0.2
      name: flux-alert-flux-alerts
    spec:
      encryptedData:
        token: test-token
      template:
        data: null
        metadata:
          annotations:
            sealedsecrets.bitnami.com/cluster-wide: "true"
          name: flux-alert-flux-alerts
